code/src/App.js

```diff
@@ -1,9 +1,66 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+
+import { THOUGHTS_URL, likeUrlFor } from 'component/Urls';
+
+import { MessageInput } from './component/MessageInput';
+import { MessageList } from './component/MessageList';
 
 export const App = () => {
+  const [messages, setMessages] = useState([]);
+
+
+  useEffect(() => {
+    fetchMessages();
+  }, []);
+
+
+  const fetchMessages = () => {
+    fetch(THOUGHTS_URL)
+      .then(res => res.json())
+      .then(data => setMessages(data))
+      .catch(err => console.log("error:", err))
+  }
+
+
+  //Post new value text input to server then fetch
+  const postMessageInput = (newMessage) => {
+    fetch(THOUGHTS_URL, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ message: newMessage })
+    })
+      .then(() => fetchMessages())
+      .catch(err => console.log("error:", err))
+  }
+
+  // Post new heart value to server then fetch
+  const postHeart = event => {
+    const messageId = event._id
+    fetch(likeUrlFor(messageId),
+      {
+        method: 'POST',
+        body: "",
+        headers: { "Content-Type": "application/json" },
+      }).then(() => {
+        onMessageLiked(messageId)
```

Comment:
Is this function (onMessageLiked) to show the likes in the browser if the server is slow to respond? I thiiink this function then needs to be placed before the fetch in that case, otherwise, it is just overwritten at once by the fetchMessages function, right?